Add .NET 9, drop .NET 6 support

vanbukin opened the pull request.

vanbukin committed:
Add .NET 9, drop .NET 6 support

vanbukin committed:
Fix README for FIDO metadata

PanSzelescik commented:
I think also [UuidVersion7Generator.cs](https://github.com/dodobrands/WebAuthn.Net/blob/main/src/WebAuthn.Net.Storage.PostgreSql/Services/Static/UuidVersion7Generator.cs) is not needed in .NET 9, because .NET 9 has [Guid.CreateVersion7()](https://learn.microsoft.com/en-us/dotnet/api/system.guid.createversion7?view=net-9.0)

vanbukin committed:
Update .NET SDKs
Reconciled .NET libs versions

vanbukin committed:
Improved docs for FIDO conformance testing.
Fix validation parameters for X509 certificate chain.

vanbukin committed:
Fix tests by configuring certificates trust-chain validation options for tests.

vanbukin committed:
Update demo project JS and CSS

vanbukin committed:
Fix links in README.md files

vanbukin committed:
Updated XMLDOC and README.md to reference the WebAuthn Level 3 specification rather than a specific draft version.

Changed PublicKeyCredentialRequestOptions to match the latest spec. Removed the attestation and attestationFormats properties. This is a breaking change.

vanbukin committed:
Update XMLDOC for AuthenticationResponse

vanbukin committed:
Update XMLDOC and contract for AuthenticatorAssertionResponse according to spec

vanbukin committed:
Update AuthenticatorAssertionResponseJSON according to specification. Fix decoding, unit tests, update XMLDOC. According to specification attestationObject property was removed. This is breaking change.

Improve logging for FIDO Conformance demo service.

Update FidoConformance.csproj to not copy .gitkeep files in metadata directories.

vanbukin committed:
Upgrade .NET SDKs and libs versions

vanbukin committed:
Cleanup solution

vanbukin committed:
Fix XMLdoc for AttestationConveyancePreference

vanbukin committed:
Update XMLDOC for AuthenticatorAttachment to reference the WebAuthn Level 3 specification and update links accordingly.

vanbukin committed:
Update XMLDOC for AuthenticatorTransport to reference WebAuthn Level 3 specification and fix links.

vanbukin committed:
Rename `PublicKeyCredentialHints` to `PublicKeyCredentialHint` to align with WebAuthn Level 3 specification. Update related references, XMLDOC, and unit tests.

vanbukin committed:
Fix XMLDOC link for PublicKeyCredentialHint to reference the final WebAuthn Level 3 specification.

vanbukin committed:
Update XMLDOC for PublicKeyCredentialType to reference WebAuthn Level 3 specification and enhance documentation

vanbukin committed:
Update XMLDOC for ResidentKeyRequirement to reference final WebAuthn Level 3 specification and fix links

vanbukin committed:
Update XMLDOC for UserVerificationRequirement to reference final WebAuthn Level 3 specification and fix links

vanbukin committed:
Update XMLDOC to reference WebAuthn Level 3 specification and fix links across `PublicKeyCredentialRequestOptionsJSON`.

vanbukin committed:
Upgrade .NET SDKs

vanbukin committed:
Set `CryptoProviderFactory.CacheSignatureProviders` to `false` on security keys, switch to `JsonWebTokenHandler`, and adjust X509 revocation mode to `Online`.

vanbukin committed:
Explicitly enable `TryAllIssuerSigningKeys` in JWT validation settings.
